plugin/src/main/java/org/opensearch/ml/cluster/MLCommonsClusterEventListener.java

```diff
@@ -24,16 +29,32 @@ public class MLCommonsClusterEventListener implements ClusterStateListener {
     private final ClusterService clusterService;
     private final MLModelManager mlModelManager;
     private final MLTaskManager mlTaskManager;
+    private final MLModelCacheHelper modelCacheHelper;
 
-    public MLCommonsClusterEventListener(ClusterService clusterService, MLModelManager mlModelManager, MLTaskManager mlTaskManager) {
+    public MLCommonsClusterEventListener(
+        ClusterService clusterService,
+        MLModelManager mlModelManager,
+        MLTaskManager mlTaskManager,
+        MLModelCacheHelper modelCacheHelper
+    ) {
         this.clusterService = clusterService;
         this.clusterService.addListener(this);
         this.mlModelManager = mlModelManager;
         this.mlTaskManager = mlTaskManager;
+        this.modelCacheHelper = modelCacheHelper;
     }
 
     @Override
     public void clusterChanged(ClusterChangedEvent event) {
+        ClusterState previousState = event.previousState();
+        ClusterState state = event.state();
+        Settings previousSettings = previousState.getMetadata().settings();
+        Settings settings = state.getMetadata().settings();
+        long previousMonitoringReqCount = ML_COMMONS_MONITORING_REQUEST_COUNT.get(previousSettings);
+        long monitoringReqCount = ML_COMMONS_MONITORING_REQUEST_COUNT.get(settings);
+        if (previousMonitoringReqCount > monitoringReqCount) {
+            modelCacheHelper.resizeMonitoringQueue(monitoringReqCount);
```

Comment:
Do we have a maximum size limitation on this queue?

Reply:
yes, the line above contains the limie "monitoringReqCount"